_posts/2017-01-17-Oredev2016.md

```diff
@@ -0,0 +1,695 @@
+Since Adam is doing this manually, I will explain you how to do it easier with docker-compose.
+
+Docker-compose will run all the containers you setup in the configuration.
+Since we work with the microservices eco-system, we will need a lot of containers.
+
+To build our image using the Dockerfile, we will add a docker dependency in our pom.xml.
+First add a property for naming your repository.
+
+```
+<docker.image.prefix>testApp</docker.image.prefix>
+```
+
+And add support for docker.
+
+```
+<plugin>
+  <groupId>com.spotify</groupId>
+  <artifactId>docker-maven-plugin</artifactId>
+  <version>0.4.13</version>
+  <executions>
+    <execution>
+      <phase>package</phase>
+        <goals>
+          <goal>build</goal>
+        </goals>
+    </execution>
+  </executions>
+  <configuration>
+    <imageName>${docker.image.prefix}/${project.artifactId}</imageName>
+    <dockerDirectory>${project.basedir}/src/main/docker</dockerDirectory>
+    <resources>
+      <resource>
+        <targetPath>/</targetPath>
+        <directory>${project.build.directory}</directory>
+        <include>${project.build.finalName}.jar</include>
+      </resource>
+    </resources>
+  </configuration>
+</plugin>
+```
+
+First we make images of our services using the Maven Docker plugin from Spotify.
+Every time Mavens builds the project, a docker image is generated.
+
+
+### Docker-compose
+To tell the containers that they have to be under the same network, we will use docker-compose.
+First we make a docker-compose.yml file and configure our microservices.
+
+```
+version: '2'
+services:
+
+  cars:
+    image: testApp/cars-service
+    ports:
+      - "8081"
+    networks:
+      - backend
+
+  gateway:
+      image: testApp/zuul-service
+      ports:
+        - "9900:9900"
+      networks:
+        - backend
+
+networks:
+  backend:
+```
+
+Variables explanation:
+
+* image
+    * The image you will be using to create the container. We will be using our microservice image
+* ports
+    * The port the service will listen to. When you want to expose your port use: "5432:5432"
+* build
+    * This will build the Dockerfile inside the same directory
+* networks as property
+    * This be the network where you want to be assigned to
+* networks
+    * This will create one or multiple networks
+
+When you want to add an eureka service, follow the above steps and configure it inside the docker-compose file.
+It is important that every container you want to communicate in between stays under the same network.
+
+<a name="docker" />
+
+# Secure systems with Docker - Emil Kvarnhammer
+
+<img class="float-image p-image" src="{{ '/img/oredev/emil.jpg' | prepend: site.baseurl }}" alt="Emil Kvarnhammer" style="width: 100px"/>
+
+[Emil Kvarnhammer](http://www.twitter.com/emilkvarnhammar) is a Security Software Engineer at TrueSec,
+a leading-edge company in IT security and development.
+He’s been involved in several security-critical projects,
```

Comment:
I notice a Microsoft Word apostrophe here, it's a good idea to find and replace all these with a regular apostrophe (').